faucet.go

```diff
@@ -0,0 +1,674 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+// faucet is an Ether faucet backed by a light client.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/ethdb/leveldb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	startKey = "StartBlock"
+)
+
+var (
+	ethRpcFlag = flag.String("ethrpc", "", "Ethereum mainnet rpc URL for ethclient to get address balance")
+	sepRpcFlag = flag.String("sepwsrpc", "", "Ethereum Sepolia websocket URL for ethclient to get swap transaction")
+	wsRpcFlag  = flag.String("wsrpc", "", "SWC websocket URL for ethclient to submit faucet or swap transactions")
+
+	apiPortFlag = flag.Int("apiport", 81, "Listener port for the HTTP API connection")
+	dataPath    = flag.String("datadir", "./es-data", "Data directory for the databases")
+
+	payoutFlag  = flag.Int("faucet.amount", 10, "Number of QKC to pay out per user request")
+	minutesFlag = flag.Int("faucet.minutes", 1440, "Number of minutes to wait between funding rounds")
+
+	accJSONFlag = flag.String("account.json", "", "Key json file to fund user requests with")
+	accPassFlag = flag.String("account.pass", "", "Decryption password to access faucet funds")
+
+	logFlag = flag.Int("loglevel", 0, "Log level to use for Ethereum and the faucet")
+)
+
+var (
+	minEthNoAuthAddrNeed = new(big.Int).Exp(big.NewInt(10), big.NewInt(16), nil)
+	ether                = new(big.Int).Exp(big.NewInt(10), big.NewInt(18), nil)
+)
+
+func main() {
+	// Parse the flags and set up the logger to print everything requested
+	flag.Parse()
+	log.SetDefault(log.NewLogger(log.NewTerminalHandlerWithLevel(os.Stderr, slog.Level(*logFlag), true)))
+
+	// Load up the account key and decrypt its password
+	blob, err := os.ReadFile(*accPassFlag)
+	if err != nil {
+		log.Crit("Failed to read account password contents", "file", *accPassFlag, "err", err)
+	}
+	pass := strings.TrimSuffix(string(blob), "\n")
+
+	path := filepath.Join(os.Getenv("HOME"), ".faucet", "keys")
+	os.RemoveAll(path)
```

Comment:
Does this mean that a new keystore file needs to be created each time the application starts?